### ISSUE TITLE
if anki folder is a symbolic link (on linux) some things don't work as expected

### ISSUE THREAD

**Issue by ijgnd** [lovac42/Fanfare#1]
Thank you very much for another great add-on!

I tested on the latest 2.1.5 on linux with only your add-on installed. The add-on doesn't work under my unusual circumstances. I have my Anki folder in a custom location. In the default (linux) location at `~/.local/share/Anki2` I have only a symbolic link to my folder. Now your add-on doesn't work as expected and sometimes throws errors. I have never had a problem before with my setup. If I remove the symlink and put the Anki folder in the default location your add-on works as expected. 

I don't know it this should be considered as a problem. If you decide to look into this on this here are some details: 

- when you start reviewing a deck you usually would hear a sound and see a picture. I only see a sound and see a placeholder for an image that wasn't found. I get no error message.
- The intermission sounds work but no picture is shown. I see a placeholder. No error message
- When I fail a card and am due for "fail feedback" I hear a sound and I get the error message `FileNotFoundError: [Errno 2] No such file or directory:  /custom/path/for/Anki2/myprofilename/collection.media/_ankiAO/2073603704/user_files/Silhouette/lapse/music.png'.`
This also happens with the "Bruce_Lee" theme.

Here is a full error message:

```python
Exception happened during processing of request from ('127.0.0.1', 49818)
Traceback (most recent call last):
  File "aqt/mediasrv.py", line 90, in send_head
FileNotFoundError: [Errno 2] No such file or directory: '/custom/path/for/Anki2/myprofilename/collection.media/_ankiAO/2073603704/user_files/Bruce_Lee/lapse/dd.png'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "socketserver.py", line 639, in process_request_thread
  File "socketserver.py", line 361, in finish_request
  File "socketserver.py", line 696, in __init__
  File "http/server.py", line 418, in handle
  File "http/server.py", line 406, in handle_one_request
  File "aqt/mediasrv.py", line 67, in do_GET
  File "aqt/mediasrv.py", line 92, in send_head
  File "http/server.py", line 473, in send_error
  File "socketserver.py", line 775, in write
BrokenPipeError: [Errno 32] Broken pipe
```

If you need other info from me just let me know and again thanks for your fantastic add-ons.

**Comment by lovac42** [comment]
Is it the same for 2.0? I believe that has to do with trying to bypass the file:/// protocol which is not allowed on qwebengine, hence the sound and no graphic.

Try changing line55 in main.py to:
   targetPath=os.path.join(os.getcwd(),'_ankiAO')

Alternatively:
If you place a copy of the theme into the user media folder or try using symbolic links for "collection.media/_ankiAO/2073603704/user_files" and disable the last monkey patch in main.py, does the picture show up?

**Comment by ijgnd** [comment]
thanks for your quick reply and help.

- I tested your add-on with Anki 2.0.52 in windows10 1803. I moved the Anki2 folder from Appdata to a different location and in the original place put a symlink with the linkshell extensions. Everything works.
- I tested with Anki 2.0.52 on linux. During startup I get an error, see [1].
- With your patch `targetPath=os.path.join(os.getcwd(),'_ankiAO')` in line 55 everything seems to work in Anki 2.1.5 in linux with a symlinked Anki folder. Then the original problem disappears.
- What you listed under "Alternatively" only partially helps on linux. I made a symlink in the media folder of my profile to the folder user_files of your add-on. Now I don't get the original error message but the picture still doesn't show.

I just posted it because now you still have the code in your head and it might have been a quick fix. Now it even works for me in my rare setup (linux+symlink). Thanks!

----

[1]
```python
Traceback (most recent call last):
  File "aqt/addons.py", line 41, in loadAddons
  File "/home/myusername/.local/share/Anki2/addons/fanfare_addon.py", line 1, in <module>
    import fanfare
  File "/home/myusername/.local/share/Anki2/addons/fanfare/__init__.py", line 1, in <module>
    from . import main
  File "/home/myusername/.local/share/Anki2/addons/fanfare/main.py", line 15, in <module>
    from .fanfare import *
  File "/home/myusername/.local/share/Anki2/addons/fanfare/fanfare.py", line 14, in <module>
    from .feedback import *
  File "/home/myusername/.local/share/Anki2/addons/fanfare/feedback.py", line 16, in <module>
    from .audFeedback import *
  File "/home/myusername/.local/share/Anki2/addons/fanfare/audFeedback.py", line 16, in <module>
    from .sound import FxWavPlayer
  File "/home/myusername/.local/share/Anki2/addons/fanfare/sound.py", line 9, in <module>
    import wave, contextlib
ImportError: No module named contextlib
```

**Comment by lovac42** [comment]
I will use os.getcwd() for the updated version than, thanks.

For 2.0, it looks like you are missing the package "contextlib" for your distribution.

**Comment by ijgnd** [comment]
In linux I use Anki 2.0.52 from the https://apps.ankiweb.net/ . This is compiled/standalone/self-containing software that doesn't rely on python packages from my distribution. 

Indeed "contextlib" is missing. Strange since it's available in Anki 2.0.52 for MacOS and Windows ...

Thanks again for your time and great add-ons in general.
